nixos/modules/system/boot/stage-2-init.sh

```diff
@@ -82,6 +84,9 @@ for o in $(cat /proc/cmdline); do
             set -- $(IFS==; echo $o)
             resumeDevice=$2
             ;;
+        init2=*)
+            set -- $(IFS==; echo $o)
+            initDaemon=$2
```

Comment:
I'm not sure, should we really make this an option of the kernel command line? Perhaps it would be better to instead provide an option like `boot.init` (name needs more thought) that would be used directly in line 217 of this file, like `exec @init@`. There are other things in this file that are already treated this way, like useHostResolvconf.

I think the only things passed via kernel params should be things that are useful to set manually when editing the grub command line before booting. IMO changing stage 2 init outside of nix config is not really that useful.